### PR TITLE
[2 of n] DCOS-6523 Add Edit Configuration modal

### PR DESCRIPTION
**Pull request by jfurrow**
Based on #123

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @daiyiwhale

File: src/js/components/modals/UpdateConfigModal.js [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    // TODO: The currentConfiguration needs to be populated with the package's
+    // current configuration.
+    // JIRA Task DCOS-6638
+    let currentConfiguration = {};
+    let {name, version} = service.get('package');
+    let icon = service.getIcons()['icon-small'];
+    let schema = service.get('config');
+
+    return (
+      <AdvancedConfig model={currentConfiguration} packageIcon={icon}
```
Comment (kennyt) [inline src/js/components/modals/UpdateConfigModal.js]: can we have one prop per line? :)

File: src/js/components/modals/UpdateConfigModal.js [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        onChange={this.handleAdvancedFormChange}
+        getTriggerSubmit={this.getConfigSubmit} />
+    );
+  }
+
+  getUpdateSuccessContent() {
+    let {servicePackage} = this.props;
+
+    let notes = Util.findNestedPropertyInObject(
+      servicePackage.get('package'),
```
Comment (kennyt) [inline src/js/components/modals/UpdateConfigModal.js]: is this not a struct? if it is, can we use a getter?

File: src/js/components/modals/UpdateConfigModal.js [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+  'handleModalClose',
+  'getConfigSubmit'
+];
+
+class UpdateConfigModal extends mixin(InternalStorageMixin, StoreMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.internalStorage_set({
+      hasFormErrors: false,
```
Comment (kennyt) [inline src/js/components/modals/UpdateConfigModal.js]: can we change these to `state`? :)

File: src/js/components/modals/UpdateConfigModal.js [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+    let serviceMetadata = this.props.service.getMetadata();
+    let {name, version} = serviceMetadata;
+
+    let {isValidated, model} = this.triggetConfigSubmit();
+
+    if (isValidated) {
+      CosmosPackagesStore.installPackage(name, version, model);
+    }
+
+    this.internalStorage_update({pendingRequest: true});
```
Comment (kennyt) [inline src/js/components/modals/UpdateConfigModal.js]: do we want to do this only if `isValidated` is true?

**Comment by jfurrow** [comment]
Closing to point to a feature branch

File: src/js/components/modals/UpdateConfigModal.js (hunk 1 above)
Comment (jfurrow) [inline src/js/components/modals/UpdateConfigModal.js]: Done! https://github.com/dcos/dcos-ui/pull/132

File: src/js/components/modals/UpdateConfigModal.js (hunk 2 above)
Comment (jfurrow) [inline src/js/components/modals/UpdateConfigModal.js]: Done! https://github.com/dcos/dcos-ui/pull/132

File: src/js/components/modals/UpdateConfigModal.js (hunk 3 above)
Comment (jfurrow) [inline src/js/components/modals/UpdateConfigModal.js]: Done! https://github.com/dcos/dcos-ui/pull/132

File: src/js/components/modals/UpdateConfigModal.js (hunk 4 above)
Comment (jfurrow) [inline src/js/components/modals/UpdateConfigModal.js]: Done! https://github.com/dcos/dcos-ui/pull/132

**Pull request closed without merging**